tutorials/examples/basic_image_array.py

```diff
@@ -0,0 +1,184 @@
+# We'll go through the same steps as before.
+
+# The following will generate additional distributions of noise.
+pink = (
+    colorednoise.powerlaw_psd_gaussian(1, img_data.shape)
+    + colorednoise.powerlaw_psd_gaussian(1, img_data.shape).T
+) * sigma
+pink_f = np.abs(np.fft.fftshift(np.fft.fft2(pink)))
+
+brown = (
+    colorednoise.powerlaw_psd_gaussian(2, img_data.shape)
+    + colorednoise.powerlaw_psd_gaussian(2, img_data.shape).T
+) * sigma
+brown_f = np.abs(np.fft.fftshift(np.fft.fft2(brown)))
+
+# Storing noises in a dictionary for reference later.
+noises = {"White": white, "Pink": pink, "Brown": brown}
+noises_f = {"White": white_f, "Pink": pink_f, "Brown": brown_f}
+
+
+# Setup some arrays to hold our data.
+#   In real use, you would probably bring your own stack of images,
+#   but we'll create some here as before.
+stack = np.zeros((3, img_data.shape[-2], img_data.shape[-1]))
+stack_f = np.zeros_like(stack)
+stack_whitened_f = np.zeros_like(stack)
+whitened_noises_f = dict()
+
+for i, name in enumerate(sorted(noises)):
+    #  Adding the different noises to our original image data.
+    stack[i] = img_data + noises[name]
+    # Compute and keep some fourier space arrays to plot later.
+    stack_f[i] = np.abs(np.fft.fftshift(np.fft.fft2(stack[i])))
+
+
+# Construct our Image and Source.
+images = Image(stack)
+img_src = ArrayImageSource(images)
+
+
+# Use ASPIRE pipeline to Whiten
+noise_estimator = WhiteNoiseEstimator(img_src)
+img_src.whiten(noise_estimator.filter)
+img_data_whitened = img_src.images(0, img_src.n).asnumpy()
+
+
+# Plot before and after whitening.
+fig, axs = plt.subplots(3, 4)
+for i, name in enumerate(sorted(noises)):
+
+    stack[i] = img_data + noises[name]
+    # Lets save the whitened fourier space
+    stack_whitened_f[i] = np.abs(np.fft.fftshift(np.fft.fft2(img_data_whitened[i])))
+
+    # and retrieve the whitened noise profile by subracting from the original signal
+    whitened_noises_f[name] = img_data_f - stack_whitened_f[i]
+
+    # and we can make some plots now
+    axs[i, 0].imshow(stack[i], cmap=plt.cm.gray)
+    axs[i, 0].set_title(f"Image with {name} Noise")
+
+    axs[i, 1].imshow(np.log(1 + stack_f[i]), cmap=plt.cm.gray)
+    axs[i, 1].set_title(f"{name} Noise Spectrum")
+
+    axs[i, 2].imshow(np.log(1 + stack_whitened_f[i]), cmap=plt.cm.gray)
+    axs[i, 2].set_title(f"Whitened {name} Noise Spection")
+
+    axs[i, 3].imshow(img_data_whitened[i], cmap=plt.cm.gray)
+    axs[i, 3].set_title(f"Image with Whitened {name} Noise")
+
+plt.show()
+
+
+# We'll also want to take a look at the spectrum power distribution.
+#  Since we just want to see the character of what is happening,
+#  I'll assume each pixel's contribution is placed at their lower left corner,
+#  and compute a crude radial profile.
+#  Code from a discussion at https://stackoverflow.com/questions/21242011/most-efficient-way-to-calculate-radial-profile.
+def radial_profile(data):
```

Comment:
Should eventually move this into the `Image` class. Not sure if this is the PR to do it, though.

Reply:
I agree. I don't know that the code I copied is what we would want to use in ASPIRE wholesale. Probably you know better, but I didn't get the impression there was a canonical computation of this. (Simple in formulas, but less so with an array in hand).

If you think this isn't appropriate for the problem, we can try to add that feature before approving this.

Reply:
Agreed. Let's stick with this here and continue the discussion in #384.